File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated/ShowWindowMessageParams.kt
@file:Suppress("FunctionName", "ClassName")
package com.sourcegraph.cody.protocol_generated

data class ShowWindowMessageParams(
  val severity: SeverityEnum? = null, // Oneof: error, warning, information
  val message: String? = null,
  val options: MessageOptions? = null,
  val items: List<String>? = null,
)

File: agent/bindings/csharp/ShowWindowMessageParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ShowWindowMessageParams
  {
    [JsonProperty(PropertyName = "severity")]
    public SeverityEnum Severity { get; set; } // Oneof: error, warning, information
    [JsonProperty(PropertyName = "message")]
    public string Message { get; set; }
    [JsonProperty(PropertyName = "options")]
    public MessageOptions Options { get; set; }
    [JsonProperty(PropertyName = "items")]
    public string[] Items { get; set; }

    public enum SeverityEnum
    {
      [EnumMember(Value = "error")]
      Error,
      [EnumMember(Value = "warning")]
      Warning,
      [EnumMember(Value = "information")]
      Information,
    }
  }
}
